core/mm/vm.c

```diff
@@ -1240,6 +1240,9 @@ TEE_Result vm_check_access_rights(const struct user_mode_ctx *uctx,
 	size_t addr_incr = MIN(CORE_MMU_USER_CODE_SIZE,
 			       CORE_MMU_USER_PARAM_SIZE);
 
+	if (!len)
```

Comment:
We aren't as forgiving in `core_is_buffer_inside()` and friends.

Prior to this change a successful result from this function guaranteed that buffer indeed was supplied. If we are to change this we need to check all the call sites to see that we don't depend on the old behaviour.